[DEBUG-5524] Document high-entropy scanner in Exception Replay Targeted Mode (#36389)
DataDog/documentation: content/en/error_tracking/backend/exception_replay.md

Reviewer: ```suggestion
snapshots become available.
```

Exception Replay has two redaction modes:

- **Strict Mode:** Redacts all values except numbers and Booleans.

> - **Targeted Mode:** Redacts known sensitive patterns such as credit card numbers, API keys, IPs, and other PII. Targeted Mode also includes a high-entropy secrets scanner that automatically redacts strings detected as having high entropy. These values appear as `[REDACTED:HIGH_ENTROPY]` in snapshots.

Reviewer: ```suggestion
- **Targeted Mode:** Redacts known sensitive patterns such as credit card numbers, API keys, IPs, and other PII. It also runs a high-entropy secrets scanner that automatically redacts likely secrets, which appear as `[REDACTED:HIGH_ENTROPY]` in snapshots.
```
